python/ray/tune/utils/trainable.py

```diff
@@ -100,6 +115,27 @@ def find_checkpoint_dir(checkpoint_path):
                     checkpoint_path))
         return os.path.normpath(checkpoint_dir)
 
+    @staticmethod
+    def find_rel_checkpoint_dir(logdir, checkpoint_path):
+        """Returns the (relative) directory name of the checkpoint.
+
+        `checkpoint_path` can be `~/ray_results/exp/
+        MyTrainable_a=1_b=2_2022-01-08_20-45-26/checkpoint_00000/xxx.pkl`.
+        or `~/ray_results/exp/MyTrainable_a=1_b=2_2022-01-08_20-45-26/
+        checkpoint_00000`.
+
+        `logdir` is a different path at `~/ray_results/exp/
+        MyTrainable_a=1_b=2_2022-01-08_20-45-51/`.
+
+        returns `checkpoint_00000`.
+
+        Raises:
+            FileNotFoundError if checkpoint directory is not found.
+        """
+        experiment_dir = os.path.dirname(os.path.dirname(logdir))
+        rel_path = os.path.relpath(checkpoint_path, experiment_dir)
+        return rel_path.split(os.sep)[1]
```

Comment:
This method is unstable and relies on specific input formatting, like trailing slashes. It also doesn't work with checkpoints from different trials or experiments (though the latter is not necessarily required).

Here are some test cases:

```

# With trailing slash --> checkpoint_0 (expected)
print(find_rel_checkpoint_dir("/tmp/exp1/trial0/", "/tmp/exp1/trial0/checkpoint_0"))

# Without trailing slash --> trial0 (should be checkpoint_0)
print(find_rel_checkpoint_dir("/tmp/exp1/trial0", "/tmp/exp1/trial0/checkpoint_0"))

# Different trial with trailing slash --> checkpoint_0 (should be ../trial1/checkpoint_0)
print(find_rel_checkpoint_dir("/tmp/exp1/trial0/", "/tmp/exp1/trial1/checkpoint_0"))

# Different trial without trailing slash --> trial1 (should be ../trial1/checkpoint_0)
print(find_rel_checkpoint_dir("/tmp/exp1/trial0", "/tmp/exp1/trial1/checkpoint_0"))

# Different experiment with trailing slash --> exp1 (should be ../../exp1/trial1/checkpoint_0)
print(find_rel_checkpoint_dir("/tmp/exp2/trial0/", "/tmp/exp1/trial1/checkpoint_0"))

# Different experiment without trailing slash --> trial1 (should be ../../exp1/trial1/checkpoint_0)
print(find_rel_checkpoint_dir("/tmp/exp2/trial0.", "/tmp/exp1/trial1/checkpoint_0"))
```

Adjusting the function like this yields the expected results:

```suggestion
        experiment_dir = os.path.dirname(logdir + os.sep)
        rel_path = os.path.relpath(checkpoint_path, experiment_dir)
        return rel_path
```

Output
```
checkpoint_0
checkpoint_0
../trial1/checkpoint_0
../trial1/checkpoint_0
../../exp1/trial1/checkpoint_0
../../exp1/trial1/checkpoint_0
```

Maybe we should add a unit test for that?

Reply:
Added a unit test. Although I think all these cases should just return checkpoint_0.

Reply:
Actually I don't think checkpoint_path and self.logdir may refer to different trials.
What observed before was an artifact of confusing test fixture set up.
So I instead removed this case and updated the test: python/ray/tune/tests/test_api.py::TrainableFunctionApiTest::testDurableTrainableClass/testDurableTrainableFunction.